consensus/misc/eip4844/eip4844.go

```diff
@@ -34,6 +34,9 @@ var (
 // if the current block contains no transactions, the excessBlobGas is updated
 // accordingly.
 func VerifyEIP4844Header(config *params.ChainConfig, parent, header *types.Header) error {
+	if header.Number.Uint64() != parent.Number.Uint64()+1 {
+		panic("bad header pair")
+	}
```

Comment:
Why number? Why not check the parent hash matches up or timestamp is 12 greater?

Reply:
The hash would be more correct, but it is way more expensive. I'm adding this check because it's cheap. Timestamp could be checked but then we'd need to update it if there will ever be a change in slot length.

Reply:
This check is just supposed to catch cases where we pass the headers in the wrong order.